builder/llpcBuilderImpl.cpp

```diff
@@ -321,6 +321,47 @@ Value* BuilderImplBase::Scalarize(
     return pResult;
 }
 
+// =====================================================================================================================
+// Helper method to scalarize in pairs a possibly vector unary operation. The callback function is called
+// with vec2 input, even if the input here is scalar.
+Value* BuilderImplBase::ScalarizeInPairs(
+    Value*                        pInValue,   // [in] Input value
+    std::function<Value*(Value*)> callback)   // [in] Callback function
+{
+    if (auto pVecTy = dyn_cast<VectorType>(pInValue->getType()))
```

Comment:
pVecTy -> pValueTy

Reply:
That loses the information that it is a vector type.

Reply:
OK. Ignore my comments.